web_api/core/test_views.py

```diff
@@ -173,6 +238,261 @@ def test_usage_billing_authentication(authed_client: Client, other_user: User) -
+    )
+    assert stripe_subscription_retrieve.call_count == 0
+    assert stripe_subscription_modify.call_count == 0
+    res = authed_client.post(
+        f"/v1/t/{account.id}/update_subscription",
+        dict(prorationTimestamp=period_start + 4 * ONE_DAY_SEC, seats=24),
+    )
+    assert res.status_code == 204
+    assert stripe_subscription_retrieve.call_count == 1
+    assert stripe_subscription_modify.call_count == 1
+    assert stripe_invoice_create.call_count == 1
+    assert stripe_invoice_pay.call_count == 1
+
+
+@pytest.mark.django_db
+def test_update_subscription_missing_customer(
+    authed_client: Client, user: User, other_user: User, mocker: Any
+) -> None:
+    fake_subscription = stripe.Subscription.construct_from(
+        dict(
+            object="subscription",
+            id="sub_Gu1xedsfo1",
+            items=dict(data=[dict(object="subscription_item", id="si_Gx234091sd2")]),
+        ),
+        "fake-key",
+    )
+
+    stripe_subscription_retrieve = mocker.patch(
+        "core.models.stripe.Subscription.retrieve", return_value=fake_subscription
+    )
+    stripe_subscription_modify = mocker.patch(
+        "core.models.stripe.Subscription.modify", return_value=fake_subscription
+    )
+    account = Account.objects.create(
+        github_installation_id=377930,
+        github_account_id=900966,
+        github_account_login=user.github_login,
+        github_account_type="User",
+        stripe_customer_id="cus_Ged32s2xnx12",
+    )
+    AccountMembership.objects.create(account=account, user=user, role="admin")
+    assert StripeCustomerInformation.objects.count() == 0
+    assert stripe_subscription_retrieve.call_count == 0
+    assert stripe_subscription_modify.call_count == 0
+    res = authed_client.post(
+        f"/v1/t/{account.id}/update_subscription",
+        dict(prorationTimestamp=1650581784, seats=24),
+    )
+    assert res.status_code == 422
+    assert res.json()["message"] == "Subscription does not exist to modify."
+    assert stripe_subscription_retrieve.call_count == 0
+    assert stripe_subscription_modify.call_count == 0
+
+
+@pytest.mark.django_db
+def test_update_subscription_not_admin(
+    authed_client: Client, user: User, other_user: User, mocker: Any
+) -> None:
+    account = Account.objects.create(
+        github_installation_id=377930,
+        github_account_id=900966,
+        github_account_login=user.github_login,
+        github_account_type="User",
+        stripe_customer_id="cus_Ged32s2xnx12",
+    )
+    AccountMembership.objects.create(account=account, user=user, role="member")
+    res = authed_client.post(
+        f"/v1/t/{account.id}/update_subscription",
+        dict(prorationTimestamp=1650581784, seats=24),
+    )
+    assert res.status_code == 403
+
+
+@pytest.mark.django_db
+def test_cancel_subscription(
+    authed_client: Client, user: User, other_user: User, mocker: Any
+) -> None:
+    ONE_DAY_SEC = 60 * 60 * 24
+    period_start = 1650581784
```

Comment:
what is this magic number?


instead can we use `date time` and convert to int from that?